src/client/dfs/dfs.c

```diff
@@ -1850,8 +1858,14 @@ dfs_cont_create(daos_handle_t poh, uuid_t *cuuid, dfs_attr_t *attr,
 	}
 
 	if (attr) {
-		dattr.da_oclass_id = attr->da_oclass_id;
-		dattr.da_dir_oclass_id = attr->da_dir_oclass_id;
+		if (attr->da_oclass_id) {
+			dattr.da_dir_oclass_id = attr->da_oclass_id;
+			dattr.da_file_oclass_id = attr->da_oclass_id;
+		}
+		if (attr->da_file_oclass_id)
+			dattr.da_file_oclass_id = attr->da_file_oclass_id;
```

Comment:
Does this have potential to break older middleware similar to what happened with hints?
Short version: I think mpifileutils is okay.
Long version:
mpifileutils is already compiled with the size including `da_hints`, so presumably `attr->da_file_oclass_id` would be allocated by mpifileutils, but would actually point to `da_hints`, which is larger, so not a memory access error. And mpifileutils doesn't use `da_hints`, so it will in practice always be NULL in mpifileutils.

Reply:
sure, but as you said mfu is fine and should be tested with the PR run.
as for other middleware, if the previous change didn't break it, then this won't :-)